circuit-examples/examples/halo2_collatz.rs

```diff
@@ -0,0 +1,46 @@
+// Given `start_number` and `output_number`, this circuit checks that applying the
+// Collatz map `collatz_iterations` many times results in `output_number`.
+//
+// The following should succeed with 6 collatz iterations:
+// `cargo run --release --example halo2_collatz_runtime 27 94`
+//
+use ark_bn254::Fr;
+use rcc::traits::BoolWire;
+use rcc_halo2::builder::{H2Wire as W, *};
+
+#[main_component]
+fn my_circuit() {
+    let collatz_iterations = 6;
+
+    let start_number = input_wire("start_number");
+    let output_number = input_wire("output_number");
```

Comment:
This also applies to the Fibonacci circuit where I missed this--`output_number` is already computed in the circuit and not used to derive any other wires, so it does not have to be an input wire.

Reply:
Concretely, we can remove `output_number` here and just call `declare_public` on `current_number` (which is`mut`) at the end of the circuit.

Reply:
Thanks! I may have been confused about how `declare_public` and `input_wire` correspond to public inputs of the circuit.

I think I answered my below question from looking at the code, but here it is anyway in case it's useful:

Q: I was thinking that in this circuit we're trying to give, as public inputs, `start_number` and `output_number`, and prove that it's a correct pair of values. So if we don't make `output_number` an input wire, does `declare_public` still make `output_number` a public input? (If so, how do I set it when I do `cargo run ...`?) Is the role of the input wires really for witness generation, rather than (what I thought) for adding a public input? 

A: Looks like the inputs are used for both generating the witness and generating the public inputs, from the line in the runtime files `let (wires, public) = generate_witnesses(inputs);`. So `declare_public` does indeed make a public input. On the other hand, `input_wire` creates wires in the circuit whose values are sufficient to generate the circuit's witness and public input. These wires could correspond to either witness values or public inputs, depending on whether we call `declare_public`.

Reply:
Yes! That's correct.